### PR TITLE
Read different types

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Make read_tensor a temple function to read different types*

### CODE REVIEW

**Comment by mvafin** [comment]
Getting error: openvino/bin/intel64/Debug/lib/libpaddlepaddle_frontend.so: undefined reference to `std::vector<float, std::allocator<float> > ngraph::frontend::InputModelPDPD::readWeight<float>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, long)'

**Review by nosovmik — APPROVED** [review]
Looks good to me

File: ngraph/frontend/paddlepaddle/src/frontend.cpp [hunk 1]
```diff
@@ -75,22 +75,64 @@ bool endsWith(const std::string &str, const std::string &suffix) {
...
         tensor.dims().cbegin(), tensor.dims().cend(), 1, std::multiplies<int64_t>());
-    // TODO: implement for other types
-    auto tensor_data = model->readWeight(var_desc->name(), tensor_length);    
-
-    std::vector<size_t> shape(tensor.dims().cbegin(), tensor.dims().cend());
-    return Constant::create(element::f32, Shape(shape), tensor_data);
+    Shape shape(tensor.dims().cbegin(), tensor.dims().cend());
+    const auto& type = TYPE_MAP[tensor.data_type()];
+    switch (type) {
+        case element::f32:
```
Comment (itikhono) [inline ngraph/frontend/paddlepaddle/src/frontend.cpp:103]: for `type `we have `.size()` method, it returns bytesize of element type
`openvino/ngraph/core/src/type/element_type.cpp` line 123

I think we can simplify all readWeights calls, or even replace them with one call 
`auto tensor_data = model->readWeight(var_desc->name(), tensor_length * type.size());`

File: ngraph/frontend/paddlepaddle/src/frontend.cpp [hunk 2]
```diff
@@ -75,22 +75,64 @@ bool endsWith(const std::string &str, const std::string &suffix) {
...
+        {
+            auto tensor_data = model->readWeight(var_desc->name(), tensor_length * sizeof(float));    
+            return Constant::create(type, shape, pdpd::reinterpret_vector<float>(tensor_data));
```
Comment (itikhono) [inline ngraph/frontend/paddlepaddle/src/frontend.cpp:106]: I think we can use this constructor of Constant 
Constant(const element::Type& type, const Shape& shape, const void* data);

and in fact pdpd :: reinterpret_vector is not needed

**Review by itikhono — APPROVED** [review]

**Pull request merged**